src/cli/common.js

```diff
@@ -305,20 +305,28 @@ function cli(api){
     function readConfigData(config) {
         var data = readConfigFile(config),
             json,
+            optionName,
+            optionValue,
+            args,
             options = {};
         if (data) {
             if (data.charAt(0) === "{") {
                 try {
                     json = JSON.parse(data);
                     data = "";
-                    for (var optionName in json) {
+                    for (optionName in json) {
                         if (json.hasOwnProperty(optionName)) {
-                            data += "--" + optionName + "=" + json[optionName].join();
+                            optionValue = json[optionName];
+                            if (typeof optionValue === "string") {
+                                optionValue = [optionValue];
+                            }
+                            data += "--" + optionName + "=" + optionValue.join(",");
                         }
                     }
                 } catch(e) {}
             }
-            options = processArguments(data.split(/[\s\n\r]+/m));
+            args = data.replace(/[\s\n\r]/g,"").split(/(?=--)/);
```

Comment:
I believe it’s more efficient to use `/[\s\n\r]+/g` instead of `/[\s\n\r]/g`, because then the replacement happens only once for each run of whitespace, instead of once for every character of whitespace.

Also, [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp#character-classes) says that `\s` includes both `\n` and `\r`, so you could simplify it further to `/\s+/g`.  :)